vispy/visuals/tests/test_linear_region.py

```diff
@@ -0,0 +1,156 @@
+
+
+@requires_application()
+def test_set_data_color_singlecolor():
+    """Test the color argument of LinearRegionVisual.set_data() method
+    using a single color"""
+
+    pos1 = [5, 42]
+    color1 = np.array([0.0, 1.0, 0.0, 0.5], dtype=np.float32)
+    color1_expected = np.array([color1, color1, color1, color1],
+                               dtype=np.float32)
+
+    with TestingCanvas() as c:
+        # Check set_data is working correctly within visual constructor
+        region = visuals.LinearRegion(pos=pos1, color=color1, parent=c.scene)
+        assert np.all(region._color == color1_expected)
+        assert np.all(region.color == color1)
+
+        region.set_data(color=list(color1))
+        assert np.all(region._color == color1_expected)
+        assert np.all(region.color == color1)
+
+        region.set_data(color=tuple(color1))
+        assert np.all(region._color == color1_expected)
+        assert np.all(region.color == color1)
+
+        # Test with different dtypes that must be converted to float32
+        region.set_data(color=color1.astype(np.float64))
+        assert np.all(region._color == color1_expected)
+        assert np.all(region.color == color1)
+
+        assert_image_approved(c.render(), 'visuals/linear_region1.png')
+
+        with raises(ValueError):
+            region.set_data(color=[1.0, 0.5, 0.5])
+
+        with raises(ValueError):
+            region.set_data(color=[color1, color1, color1])
+
+        region.parent = None
+
+
+@requires_application()
+def test_set_data_color_multicolor():
+    """Test the color argument of LinearRegionVisual.set_data() method
+    using multiple colors"""
+
+    pos2 = [5, 42, 80]
+    color2 = np.array([[0.0, 1.0, 0.0, 0.5],
+                       [1.0, 0.0, 0.0, 0.75],
+                       [0.0, 0.0, 1.0, 1.0]], dtype=np.float32)
+    color2_expected = np.array([color2[0], color2[0],
+                                color2[1], color2[1],
+                                color2[2], color2[2]],
+                               dtype=np.float32)
+
+    with TestingCanvas() as c:
+        # Check set_data is working correctly within visual constructor
+        region = visuals.LinearRegion(pos=pos2, color=color2, parent=c.scene)
+        assert np.all(region._color == color2_expected)
+        assert np.all(region.color == color2)
+
+        region.set_data(color=list(color2))
+        assert np.all(region._color == color2_expected)
+        assert np.all(region.color == color2)
+
+        region.set_data(color=tuple(color2))
+        assert np.all(region._color == color2_expected)
+        assert np.all(region.color == color2)
+
+        # Test with different dtypes that must be converted to float32
+        region.set_data(color=color2.astype(np.float64))
+        assert np.all(region._color == color2_expected)
+        assert np.all(region.color == color2)
+
+        assert_image_approved(c.render(), 'visuals/linear_region2.png')
+
+        region.parent = None
```

Comment:
My purpose is to ensure that both `list`, `tuple` and numpy arrays are accepted as `color` argument and have the same result.
Do you think it can be ignored?